sips/sip-011/sip-011-catastrophic-failure-recovery.md

```diff
@@ -0,0 +1,464 @@
+Therefore, the kinds of catastrophic failures that necessitate hard forks are
+artifacts of a design bug in the Stacks blockchain that renders it unsustainable
+or unable to operate as intended.
+
+Some examples of when a hard fork is the appropriate course of action include,
+but are not limited to:
+
+* The underlying burnchain is crashing or is rendered insufficiently reliable
+  for the Stacks blockchain to keep making progress.
+* The discovery of a cryptographic break in the signing/verifying algorithms
+  used by the Stacks blockchain makes all funds vulnerable to theft.
+* A severe bug in the consensus-critical code-paths leads to an irreconcilable
+  chain split, where it is ambiguous as to which tip is the "true" tip.
+
+Importantly, the following problems are _not_ worthy of a hard fork:
+
+* Theft of funds due to buggy but correctly-executed Clarity code.
+* Loss or theft of funds due to user error.
+* Different opinions on the design of the Stacks blockchain.
+* Different opinions on the design of an already-deployed smart contract.
+
+Depending on the severity of the underlying bug, the procedure for executing a
+hard fork presented here should be considered preliminary and incomplete.  At a
+minimum, the following steps will be followed:
+
+1. An announcement describing the situation will be sent via signed email from
+announce@stacks.org
+2. If at all possible, one of the above procedures will be executed to bring the
+blockchain network back online in a degraded state at the latest point of
+execution before where it had failed (as measured by burnchain block height).
+3. If mining can be resumed, the soft fork procedure will be executed to vote
+for a "flag day" that will cause this version of the blockchain to crash
+permanently at the end of the voting period.  If mining cannot be resumed, then
+this step is skipped.
+4. A new version of the node software will be released via the critical bugfix
+point-release procedure.  The node software will resume from the old chainstate
+at the last block mined before the catastrophic failure was manifested (as
+determined by the burnchain block height).
+
+Step 3 should be executed if possible, because it gives the network a chance to
+vote on the new rules.
+
+Note that in this procedure, at least two emails from announce@stacks.org will
+be sent -- one in step 1 and one in step 4.  However, given the severity of any
+bug that can only be resolved with a hard fork, there may be many more
+communications sent.
+
+# Related Work
+
+This section is a work in process.
+
+Bitcoin has a codified process for soft fork votes.  It should be cited here.
+
+Tezos has an even more extreme version of soft-forking than we support, but it's
+still a related concept to be cited.
+
+Ethereum is a good example of how _not_ to do hard forks -- they just happen
+willy-nilly once enough "core developers" are on board.
+
+# Activation
+
+This section is a work in progress.  Feedback is welcome.
```

Comment:
Neither options are binding nor enforceable. 

Option 1 is no different than the entities (or their verified representatives) publicly expressing support as a comment on this PR?

I like the idea of capturing the support on-chain. Perhaps we can combine the two options by having a smart-contract that determines activation based on contract-calls from a set of well-known, whitelisted addresses representing the entities here?

Reply:
This SIP mainly affects the behaviors of Stacks Core Developers and the Stacks Foundation, since it codifies their response to catastrophic failures.  It's meant to be informational to all other ecosystem participants -- i.e. to make sure they know what to expect in the event of a catastrophic failure.  So, that makes me think that neither of these are good options.  I think it might be sufficient to just have approval from the Stacks Core Developers who will be responsible for fixing these problems if they come up.  Of course, other ecosystem participants are free to do what they like -- nothing about this SIP can be legally or programmatically enforced.

Reply:
I went ahead and dropped options 1 and 2 in 3832ae2 in favor of just ratifying this SIP by having the Stacks Core Developers who will be responsible for running these procedures sign off on it.  One way or another, other ecosystem participants are going to be looking to these folks (and their employers) for guidance if the blockchain encounters a catastrophic error, so I don't think it's necessary to get their sign-off (they already expect us to help them).